home/migrations/0004_test_data_book_subsubject_subject.py

```diff
@@ -0,0 +1,84 @@
+from django.db import migrations, transaction
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('home', '0003_book_sub_subject_subject'),
+    ]
+
+    def generate_data(apps, schema_editor):
+        from home.models import Subject, Sub_Subject, Book
+        subjects_test_data = [
+            ('Bible'),
+            ('Biology'),
+            ('Chemistry'),
+            ('English'),
+            ('Geography'),
+            ('Literature'),
+            ('History'),
+            ('Hebrew'),
+            ('Math'),
+            ('Physics')
+        ]
+        sub_subjects_test_data = [
+            ('Gensis', 'Bible'),
+            ('Kings in the Bible', 'Bible'),
+            ('Prophecies', 'Bible'),
+            ('Anatomy', 'Biology'),
+            ('Genetics', 'Biology'),
+            ('Cell Biology', 'Biology'),
+            ('Atomic Structure', 'Chemistry'),
+            ('Units and Measurement', 'Chemistry'),
+            ('Periodic Table', 'Chemistry'),
+            ('Essays', 'English'),
+            ('Grammer', 'English'),
+            ('English - Oral', 'English'),
+            ('Physical Geography', 'Geography'),
+            ('Human Geography', 'Geography'),
+            ('Atmosphere', 'Geography'),
+            ('Medieval', 'Literature'),
+            ('17th Century', 'Literature'),
+            ('20th Century', 'Literature'),
+            ('Writing', 'Hebrew'),
+            ('Text analysis', 'Hebrew'),
+            ('Grammer', 'Hebrew'),
+            ('Algebra', 'Math'),
+            ('Geometry ', 'Math'),
+            ('Probability', 'Math'),
+            ('Classical mechanics', 'Physics'),
+            ('Thermodynamics ', 'Physics'),
+            ('Quantum mechanics', 'Physics'),
+            ('Ancient history‎', 'History'),
+            ('Roman period ', 'History'),
+            ('The Middle Ages', 'History'),
+        ]
+        book_test_data = [
+            ('The Atlas of the Human Body', 'Biology'),
+            ('Genesis', 'Bible'),
+            ('Exodus', 'Bible'),
+            ('Judges', 'Bible'),
+            ('All My Sons', 'Literature'),
+            ('1948', 'Literature'),
+            ('antigona', 'Literature'),
+            ('Sapiens: A Brief History of Humankind', 'History'),
+            ('Modern Hebrew for Beginners', 'Hebrew'),
+            ('Benny-Goren', 'Math'),
+            ('Fundamentals of Physics', 'Physics')
+        ]
+
+        with transaction.atomic():
```

Comment:
This is quite an inefficient way to load this data, since you are running a database query fro every single Sub_Subject and Book record you're adding.

This could potentially be optimized by nesting that test data structures, so that you can have an outer loop that added the Subject records and an inner loop that added the Sub_Subject and Book records.

In any case will merge for now in the interest of time.